### PR TITLE
Bug 1856067 - Support provisioning profiles in hardened signing

### PR DESCRIPTION
**Pull request by hneiva**

### CODE REVIEW

**Comment by hneiva** [comment]
On second thought, I'll change how the provisioning profile destination is configured.

File: iscript/tests/test_hardened_sign.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    raise IScriptError("fail_async exception")
+
+
+hs_config = [{"deep": True, "runtime": True, "force": True, "entitlements": "https://foo.bar", "globs": ["/"]}]
+
+
+@pytest.mark.asyncio
+async def test_download_signing_resources(mocker):
+    mocker.patch.object(hs, "retry_async", new=noop_async)
+    await hs.download_signing_resources(hs_config, Path("fakefolder"))
```
Comment (gbrownmozilla) [inline iscript/tests/test_hardened_sign.py:33]: Could assert on the return value here.

File: iscript/tests/test_hardened_sign.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+def test_check_globs():
+    globs = ["doesntexist/*", "/*"]
+    hs.check_globs(TEST_DATA_DIR, globs)
```
Comment (gbrownmozilla) [inline iscript/tests/test_hardened_sign.py:38]: Could change check_globs to return something, then assert on that here.

File: iscript/tests/test_hardened_sign.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    # Source file doesn't exist
+    with pytest.raises(IScriptError):
+        hs.copy_provisioning_profile(pprofile, tmpdir, config)
+
+
+def test_build_sign_command(tmpdir):
+    file_map = {
+        hs_config[0]["entitlements"]: "filepath"
+    }
+    hs.build_sign_command(tmpdir, "12345identity", "keychainpath", hs_config[0], file_map)
```
Comment (gbrownmozilla) [inline iscript/tests/test_hardened_sign.py:66]: Assert on return value? Check more cases?

**Review by gbrownmozilla — APPROVED** [review]
lgtm. Just a few suggestions for more tests, for follow-up if you like.

**Comment by hneiva** [comment]
I plan on adding more functionality to the hardened signing behavior soon. Will improve testing as we extend functionality.

**Pull request merged**